File: backend/internal/dtos/machine/updateMachine.dto.go
package dtos

type UpdateMachineType struct {
	Name           *string `json:"name"`
	Description    *string `json:"description"`
	Photo          *string `json:"photo"`
	CategoryKey    *string `json:"categoryKey"`
	TrackingType   *string `json:"trackingType"`
	RequiresWeight *bool   `json:"requiresWeight"`
}

File: admin/src/main/java/com/pedro/fitchaadmin/machine/dtos/UpdateMachineDTO.java
package com.pedro.fitchaadmin.machine.dtos;

import java.util.List;

public record UpdateMachineDTO(
        String slug,
        String name,
        String description,
        String photo,
        boolean removePhoto,
        String categoryKey,
        String substitutionGroup,
        String trackingType,
        boolean requiresWeight,
        List<String> aliases
) {
}
